File: BinarySearch.java
import java.util.Scanner;
public class BinarySearch {
    public static int binarySearch(int[] arr, int sree) {
        int low = 0, high = arr.length - 1;
        while (low <= high) {
            int mid = (low + high) / 2;

            if (arr[mid] == sree) 
                return mid;
            else if (arr[mid] < sree)
                low = mid + 1;
            else 
                high = mid - 1;
        }
        return -1; 
    }
    public static void main(String[] args) {
        int[] arr = {2, 4, 6, 8, 10}; 
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a number to search: ");
        int sree = scanner.nextInt();
        int result = binarySearch(arr, sree);
        if (result != -1)
            System.out.println("sree is  at kadapa " + result);
        else
            System.out.println("sree not found");
    }
}

File: Binary search.c
#include <stdio.h>

// Binary Search Function
int binarySearch(int arr[], int low, int high, int target) {
    while (low <= high) {
        int mid = (low + high) / 2;

        if (arr[mid] == target) 
            return mid;
        else if (arr[mid] < target) 
            low = mid + 1;
        else 
            high = mid - 1;
    }

    return -1; 
}

int main() {
    int arr[] = {1, 3, 5, 7, 9}; 
    int size = sizeof(arr) / sizeof(arr[0]);
    int target;

    printf("Enter a number to search: ");
    scanf("%d", &target);

    int result = binarySearch(arr, 0, size - 1, target);

    if (result != -1)
        printf("Number found at index %d\n", result);
    else
        printf("Number not found\n");

    return 0;
}

